localstack/services/awslambda/provider.py

```diff
@@ -251,14 +720,1677 @@ def invoke(
+        function_name, qualifier = api_utils.get_name_and_qualifier(
+            function_name, qualifier, context.region
+        )
+
+        if (
+            maximum_event_age_in_seconds is None
+            and maximum_retry_attempts is None
+            and destination_config is None
+        ):
+            raise InvalidParameterValueException(
+                "You must specify at least one of error handling or destination setting.",
+                Type="User",
+            )
+
+        fn = state.functions.get(function_name)
+        if not fn:
+            raise ResourceNotFoundException("TODO: Where fn?")
+        if qualifier and not (qualifier in fn.aliases or qualifier in fn.versions):
+            raise ResourceNotFoundException("The function doesn't exist.", Type="User")
+
+        qualifier = qualifier or "$LATEST"
+
+        config = fn.event_invoke_configs.get(qualifier)
+        if not config:
+            raise ResourceNotFoundException("TODO: config not found")
+
+        if destination_config:
+            self._validate_destination_config(state, function_name, destination_config)
+
+        optional_kwargs = {
+            k: v
+            for k, v in {
+                "destination_config": destination_config,
+                "maximum_retry_attempts": maximum_retry_attempts,
+                "maximum_event_age_in_seconds": maximum_event_age_in_seconds,
+            }.items()
+            if v is not None
+        }
+
+        new_config = dataclasses.replace(
+            config, last_modified=api_utils.generate_lambda_date(), **optional_kwargs
+        )
+        fn.event_invoke_configs[qualifier] = new_config
+
+        return FunctionEventInvokeConfig(
+            LastModified=datetime.datetime.strptime(
+                new_config.last_modified, api_utils.LAMBDA_DATE_FORMAT
+            ),
+            FunctionArn=api_utils.qualified_lambda_arn(
+                function_name, qualifier or "$LATEST", context.account_id, context.region
+            ),
+            DestinationConfig=new_config.destination_config,
+            MaximumEventAgeInSeconds=new_config.maximum_event_age_in_seconds,
+            MaximumRetryAttempts=new_config.maximum_retry_attempts,
+        )
+
+    # =======================================
+    # ======  Layer & Layer Versions  =======
+    # =======================================
+
+    def publish_layer_version(
+        self,
+        context: RequestContext,
+        layer_name: LayerName,
+        content: LayerVersionContentInput,
+        description: Description = None,
+        compatible_runtimes: CompatibleRuntimes = None,
+        license_info: LicenseInfo = None,
+        compatible_architectures: CompatibleArchitectures = None,
+    ) -> PublishLayerVersionResponse:
+        ...
+
+    def get_layer_version(
+        self, context: RequestContext, layer_name: LayerName, version_number: LayerVersionNumber
+    ) -> GetLayerVersionResponse:
+        ...
```

Comment:
nit: i think we should use `pass`. we've discussed using ellipses in anything else than stub files before: https://github.com/localstack/localstack/pull/5478#discussion_r807184504